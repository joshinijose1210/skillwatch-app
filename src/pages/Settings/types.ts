export interface DomainData {
    name: string;
    id: number;
    organisationId: number;
    error?: string;
    isDomainUsed?: boolean;
}

export interface SettingStatus {
    isManagerReviewMandatory: boolean;
    isAnonymousSuggestionAllowed: boolean;
    isBiWeeklyFeedbackReminderEnabled: boolean;
}
