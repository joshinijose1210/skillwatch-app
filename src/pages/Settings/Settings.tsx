import { Loader, PageContent } from '@components';
import ListHeader from '@components/reusableComponents/ListHeader';
import { NoteMessage } from '@constants/data';
import { Button, Modal, Text } from '@medly-components/core';
import { EventNoteIcon, DeleteForeverIcon } from '@medly-components/icons';
import { AddTeamButtonDiv } from '@pages/FirstTeam/FirstTeam.styled';
import { ClearButton } from '@pages/ReviewCycleActionItems/ReviewCycleActionItems.styled';
import { CancelButton } from '@pages/SelfReview/SelfReviewForm/SelfReviewForm.styled';
import {
    Container,
    DomainAndDeleteButtonDiv,
    LeftWrapper,
    NoteBox,
    StyleButtonWrapper,
    StyledAddDomain,
    TipHeader,
    StyledText,
    ToggleWrapper,
    GeneralSettingsContainer,
    StyledTextField
} from './Settings.style';
import { useSettings } from './useSettings';
import { Step, StepsList } from '@pages/FeedbackForm/FeedbackTipsDesc.styles';
import { StyledModalHeader, StyledModalContent, StyledModalActions, StyledToggle, StyledTipsTitle } from '@common';

export const Settings = () => {
    const {
        actionFrom,
        handleRemoveField,
        inputFields,
        handleAddAnotherDomain,
        handleInputChange,
        existDomain,
        addNewDomainLoading,
        buttonDisabled,
        isLoading,
        isModalOpen,
        setModalOpen,
        deleteId,
        hasEditPermission,
        settingStatus,
        handleDisableDomain,
        handleToggleClick,
        handleSubmit,
        settingLading,
        updateSettingLoading,
        isManagerReviewStart
    } = useSettings();

    if (isLoading || settingLading) {
        return <Loader />;
    }
    return (
        <PageContent>
            <ListHeader title="Settings" />
            <GeneralSettingsContainer>
                <LeftWrapper>
                    <StyledText hasTopMargin={false} textVariant="h4" textWeight="Regular">
                        General Settings
                    </StyledText>
                    <ToggleWrapper>
                        <StyledToggle
                            label="Manager Response Mandatory for Manager Review: "
                            data-testid="toggleManagerReview"
                            disabled={isManagerReviewStart}
                            checked={settingStatus.isManagerReviewMandatory}
                            onClick={() => handleToggleClick('isManagerReviewMandatory')}
                        />
                    </ToggleWrapper>
                    <ToggleWrapper>
                        <StyledToggle
                            label="Employees can submit suggestions Anonymously: "
                            data-testid="toggleAnonymousSuggestion"
                            checked={settingStatus.isAnonymousSuggestionAllowed}
                            onClick={() => handleToggleClick('isAnonymousSuggestionAllowed')}
                        />
                    </ToggleWrapper>
                    <ToggleWrapper>
                        <StyledToggle
                            label="Send Bi-weekly feedback reminders to slack channel: "
                            data-testid="isBiWeeklyFeedbackReminderEnabled"
                            checked={settingStatus.isBiWeeklyFeedbackReminderEnabled}
                            onClick={() => handleToggleClick('isBiWeeklyFeedbackReminderEnabled')}
                        />
                    </ToggleWrapper>
                </LeftWrapper>
                <NoteBox>
                    <TipHeader>
                        <EventNoteIcon iconColor="#000" size="S" />
                        <StyledTipsTitle>Notes</StyledTipsTitle>
                    </TipHeader>
                    <StepsList>
                        <Step>
                            <Text style={{ fontSize: '14px' }}>
                                <strong>Manager Response Mandatory:</strong> When enabled, managers must provide both a review and a rating
                                during the review process. If disabled, only a rating is required during the review process.
                            </Text>
                        </Step>
                        <Step>
                            <Text style={{ fontSize: '14px' }}>
                                <strong>Anonymous Suggestions:</strong> When enabled, employees have the option to submit their
                                suggestions/feedback to Suggestion Box anonymously, ensuring privacy and encouraging open feedback.
                            </Text>
                        </Step>
                        <Step>
                            <Text style={{ fontSize: '14px' }}>
                                <strong>Bi-weekly reminders:</strong> When enabled, reminders will be sent to the integrated slack channel
                                every alternate Monday at 12 pm to share feedback and keep 1:1 conversations active.
                            </Text>
                        </Step>
                        {/* TODO: uncomment after backend changes are deployed */}
                        {/* <Step>
                            <Text style={{ fontSize: '14px' }}>
                                <strong>Bi-weekly reminders:</strong> When enabled, reminders will be sent to the integrated slack channel
                                every alternate Monday at 12 pm to share feedback and keep 1:1 conversations active.
                            </Text>
                        </Step> */}
                    </StepsList>
                </NoteBox>
            </GeneralSettingsContainer>

            <Container>
                <LeftWrapper>
                    <StyledText hasTopMargin={false} textVariant="h4" textWeight="Regular">
                        Allowed Domains
                    </StyledText>
                    {existDomain &&
                        existDomain.length > 0 &&
                        existDomain.map((domain: any, index: number) => {
                            return (
                                <DomainAndDeleteButtonDiv key={domain.id}>
                                    <StyledTextField
                                        disabled={true}
                                        variant="outlined"
                                        name="name"
                                        label="Domain Name"
                                        data-testid="domain"
                                        value={domain.name}
                                        onChange={(e: React.ChangeEvent<HTMLInputElement>) => handleInputChange(e, domain.id)}
                                    />
                                    {index !== 0 && hasEditPermission && (
                                        <ClearButton
                                            disabled={domain.isDomainUsed}
                                            title={
                                                domain.isDomainUsed
                                                    ? 'Employees are added from the mentioned domain name. So it cannot be deleted.'
                                                    : ''
                                            }
                                            onClick={() => handleDisableDomain(domain.id)}
                                        >
                                            <DeleteForeverIcon
                                                data-testid="deleteDomainButton"
                                                disabled={domain.isDomainUsed}
                                                iconColor="red"
                                            />
                                        </ClearButton>
                                    )}
                                </DomainAndDeleteButtonDiv>
                            );
                        })}
                    {hasEditPermission &&
                        inputFields &&
                        inputFields.length > 0 &&
                        inputFields.map((input: any) => {
                            return (
                                <DomainAndDeleteButtonDiv key={input.id}>
                                    <StyledTextField
                                        disabled={false}
                                        type="text"
                                        variant="outlined"
                                        name="name"
                                        label="Domain Name"
                                        value={input.name}
                                        onChange={(e: any) => handleInputChange(e, input.id)}
                                        errorText={input.error}
                                        data-testid="inputDomainName"
                                    />

                                    <ClearButton data-testid="clearInput" type="button" onClick={() => handleRemoveField(input.id)}>
                                        <DeleteForeverIcon iconColor="red" />
                                    </ClearButton>
                                </DomainAndDeleteButtonDiv>
                            );
                        })}
                    {hasEditPermission && (
                        <>
                            <AddTeamButtonDiv>
                                <StyledAddDomain textVariant="h4" onClick={handleAddAnotherDomain} textAlign="center">
                                    + Add Domain
                                </StyledAddDomain>
                            </AddTeamButtonDiv>
                        </>
                    )}
                    <StyleButtonWrapper>
                        <Button
                            disabled={buttonDisabled}
                            isLoading={(actionFrom === 'add' && addNewDomainLoading) || updateSettingLoading}
                            variant="solid"
                            onClick={handleSubmit}
                        >
                            Save
                        </Button>
                    </StyleButtonWrapper>
                </LeftWrapper>

                <NoteBox>
                    <TipHeader>
                        <EventNoteIcon iconColor="#000" size="S" />
                        <StyledTipsTitle>Notes</StyledTipsTitle>
                    </TipHeader>
                    <Text style={{ paddingTop: '10px', fontSize: '14px' }}>
                        <strong>Allowed Domain: </strong>
                        {NoteMessage}
                    </Text>
                </NoteBox>
            </Container>

            <Modal open={isModalOpen} onCloseModal={() => setModalOpen(false)}>
                <StyledModalHeader>
                    <Text textVariant="h2">Confirm</Text>
                </StyledModalHeader>
                <StyledModalContent>
                    <Text textVariant="body1" textWeight="Medium">
                        Are you sure you want to delete the domain?
                    </Text>
                </StyledModalContent>
                <StyledModalActions>
                    <CancelButton
                        isLoading={actionFrom === 'delete' && addNewDomainLoading}
                        onClick={() => handleRemoveField(deleteId, 'existDomain')}
                        variant="outlined"
                    >
                        Yes
                    </CancelButton>

                    <Button onClick={() => setModalOpen(false)} variant="outlined">
                        No
                    </Button>
                </StyledModalActions>
            </Modal>
        </PageContent>
    );
};
