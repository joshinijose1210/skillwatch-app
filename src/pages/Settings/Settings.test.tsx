import { RTKRender, cleanup, fireEvent, screen, waitFor } from '@utils/test-utils';
import { Settings } from './Settings';
import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import { SettingsMockData } from '@mocks/settings';
import { Loader } from '@components';

const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });

beforeEach(mock.reset);

afterEach(cleanup);

describe('Settings page', () => {
    it('should take snapshot of loader', () => {
        const { container } = RTKRender(<Settings />);
        const { container: loaderContainer } = RTKRender(<Loader />);
        expect(container.innerHTML).toMatch(loaderContainer.innerHTML);
    });

    it('should take snapshot of loaded page and add new domain', async () => {
        mock.onGet('https://dummy-api.example.com/api/organisation/general-settings/?organisationId=1').replyOnce(200, {
            ...SettingsMockData.organisationData
        });
        mock.onGet('https://dummy-api.example.com/api/organisation/domains/?id=1').replyOnce(200, SettingsMockData.domains);

        mock.onGet('https://dummy-api.example.com/api/review-cycle/timeline/?organisationId=1&reviewToId=72').replyOnce(200, {
            ...SettingsMockData.timelineData
        });
        mock.onPost('https://dummy-api.example.com/api/organisation/domains/').replyOnce(200);

        const { container } = RTKRender(<Settings />, { initialState: SettingsMockData.initialState });
        expect(await screen.findByText(/Employees can submit suggestions Anonymously:/i)).toBeInTheDocument();

        expect(container).toMatchSnapshot();

        fireEvent.click(screen.getByText('+ Add Domain'));

        const addDomainInput = await screen.findByTestId<HTMLInputElement>('inputDomainName');
        fireEvent.change(addDomainInput, { target: { value: 'invalid email' } });
        expect(await screen.findByText('Please enter valid domain name')).toBeInTheDocument();
        fireEvent.click(screen.getByTestId('clearInput'));
        fireEvent.click(screen.getByText('+ Add Domain'));
        const addDomainInput2 = await screen.findByTestId<HTMLInputElement>('inputDomainName');

        fireEvent.change(addDomainInput2, { target: { value: 'newdomain.com' } });
        await waitFor(() => expect(addDomainInput2).toHaveValue('@newdomain.com'));
        const saveButton = await screen.findByText('Save');
        expect(saveButton).toBeEnabled();
        fireEvent.click(saveButton);
        expect(await screen.findByText('Allowed Domain list updated successfully')).toBeInTheDocument();
    });

    it('should delete a domain if it is not used', async () => {
        mock.onGet('https://dummy-api.example.com/api/organisation/domains/?id=1').replyOnce(200, SettingsMockData.domains);
        mock.onGet('https://dummy-api.example.com/api/organisation/?id=21').replyOnce(200, {
            ...SettingsMockData.companyData
        });
        mock.onGet('https://dummy-api.example.com/api/organisation/general-settings/?organisationId=1').replyOnce(200, {
            ...SettingsMockData.organisationData
        });

        mock.onGet('https://dummy-api.example.com/api/review-cycle/timeline/?organisationId=1&reviewToId=72').replyOnce(200, {
            ...SettingsMockData.timelineData
        });
        mock.onPost('https://dummy-api.example.com/api/organisation/domains/').replyOnce(200);

        RTKRender(<Settings />, { initialState: SettingsMockData.initialState });
        await screen.findByDisplayValue('@scalereal.com');
        const domains = await screen.findAllByText('Domain Name');
        const deleteDomainButtons = await screen.findAllByTestId('deleteDomainButton');
        await waitFor(() => expect(domains).toHaveLength(3));
        await waitFor(() => expect(deleteDomainButtons).toHaveLength(2)); // since atleast one domain should be in the list that cannot be deleted

        fireEvent.click(deleteDomainButtons[1]);

        expect(await screen.findByText('Are you sure you want to delete the domain?')).toBeInTheDocument();

        const closeIcon = document.getElementById('medly-modal-close-button') as HTMLElement;
        expect(closeIcon).toBeInTheDocument();
        fireEvent.click(closeIcon);

        fireEvent.click(deleteDomainButtons[1]);
        fireEvent.click(await screen.findByText('No'));

        fireEvent.click(deleteDomainButtons[1]);

        expect(await screen.findByText('Are you sure you want to delete the domain?')).toBeInTheDocument();
        fireEvent.click(await screen.findByText('Yes'));
        expect(await screen.findByText('Allowed Domain list updated successfully')).toBeInTheDocument();
    });

    it('should toggle anonymous suggestion send, manager review response mandatory and bi-weekly reminders', async () => {
        RTKRender(<Settings />, { initialState: SettingsMockData.initialState });
        mock.onGet('https://dummy-api.example.com/api/organisation/?id=21').replyOnce(200, {
            ...SettingsMockData.companyData
        });
        mock.onGet('https://dummy-api.example.com/api/organisation/general-settings/?organisationId=1').replyOnce(200, {
            ...SettingsMockData.organisationData
        });

        mock.onGet('https://dummy-api.example.com/api/review-cycle/timeline/?organisationId=1&reviewToId=72').replyOnce(200, {
            ...SettingsMockData.timelineData
        });
        mock.onPut('https://dummy-api.example.com/api/organisation/general-settings/').replyOnce(200, {});
        const toggleManagerReview = await screen.findByTestId('toggleManagerReview');
        const toggleAnonymousSuggestion = await screen.findByTestId('toggleAnonymousSuggestion');
        const toggleBiWeeklyFeedbackReminderEnabled = await screen.findByTestId('isBiWeeklyFeedbackReminderEnabled');

        expect(toggleManagerReview).not.toBeChecked();
        expect(toggleAnonymousSuggestion).not.toBeChecked();
        expect(toggleBiWeeklyFeedbackReminderEnabled).toBeChecked();

        fireEvent.click(toggleAnonymousSuggestion);
        fireEvent.click(toggleManagerReview);
        fireEvent.click(toggleBiWeeklyFeedbackReminderEnabled);

        expect(toggleManagerReview).toBeChecked();
        expect(toggleAnonymousSuggestion).toBeChecked();
        expect(toggleBiWeeklyFeedbackReminderEnabled).not.toBeChecked();

        const saveButton = await screen.findByText('Save');
        expect(saveButton).toBeEnabled();

        fireEvent.click(saveButton);

        expect(await screen.findByText('Settings updated successfully')).toBeInTheDocument();
    });

    it('should fail to delete a domain', async () => {
        mock.onGet('https://dummy-api.example.com/api/organisation/general-settings/?organisationId=1').replyOnce(200, {
            ...SettingsMockData.organisationData
        });
        mock.onGet('https://dummy-api.example.com/api/organisation/domains/?id=1').replyOnce(200, SettingsMockData.domains);

        mock.onGet('https://dummy-api.example.com/api/review-cycle/timeline/?organisationId=1&reviewToId=72').replyOnce(200, {
            ...SettingsMockData.timelineData
        });
        mock.onPost('https://dummy-api.example.com/api/organisation/domains/').replyOnce(404);
        RTKRender(<Settings />, { initialState: SettingsMockData.initialState });
        expect(await screen.findByText(/Employees can submit suggestions Anonymously:/i)).toBeInTheDocument();
        fireEvent.click(screen.getByText('+ Add Domain'));
        const addDomainInput = await screen.findByTestId<HTMLInputElement>('inputDomainName');
        fireEvent.change(addDomainInput, { target: { value: 'newdomain.com' } });
        const saveButton = await screen.findByText('Save');
        expect(saveButton).toBeEnabled();
        fireEvent.click(saveButton);

        expect(await screen.findByText(/Unable to update allowed domains list./i)).toBeInTheDocument();
    });

    it('should fail to update settings with a proper error message', async () => {
        RTKRender(<Settings />, { initialState: SettingsMockData.initialState });
        mock.onGet('https://dummy-api.example.com/api/organisation/?id=21').replyOnce(200, {
            ...SettingsMockData.companyData
        });
        mock.onGet('https://dummy-api.example.com/api/organisation/general-settings/?organisationId=1').replyOnce(200, {
            ...SettingsMockData.organisationData
        });

        mock.onGet('https://dummy-api.example.com/api/review-cycle/timeline/?organisationId=1&reviewToId=72').replyOnce(200, {
            ...SettingsMockData.timelineData
        });
        mock.onPut('https://dummy-api.example.com/api/organisation/general-settings/').replyOnce(400, {});
        const toggleManagerReview = await screen.findByTestId('toggleManagerReview');
        const toggleAnonymousSuggestion = await screen.findByTestId('toggleAnonymousSuggestion');

        expect(toggleManagerReview).not.toBeChecked();
        expect(toggleAnonymousSuggestion).not.toBeChecked();

        fireEvent.click(toggleAnonymousSuggestion);
        fireEvent.click(toggleManagerReview);
        expect(toggleManagerReview).toBeChecked();
        expect(toggleAnonymousSuggestion).toBeChecked();
        const saveButton = await screen.findByText('Save');
        expect(saveButton).toBeEnabled();
        fireEvent.click(saveButton);

        expect(await screen.findByText('Unable to update settings.')).toBeInTheDocument();
    });
});
