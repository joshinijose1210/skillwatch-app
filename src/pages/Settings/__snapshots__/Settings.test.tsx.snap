// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings page should take snapshot of loaded page and add new domain 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Settings
        </h3>
      </div>
    </div>
    <div
      class="sc-1porr9n-0 sc-1porr9n-10 gWazlW dhgDFT"
    >
      <div
        class="sc-1porr9n-1 cnhRJl"
      >
        <h4
          class="sc-1ntgl3l-0 fgmHgf sc-1porr9n-8 iWBSaF"
          textalign="initial"
          textcolor="inherit"
          textvariant="h4"
          textweight="Regular"
        >
          General Settings
        </h4>
        <div
          class="sc-tkiih6-29 sc-1porr9n-9 fWtWBE ldQZxM"
        >
          <div
            class="sc-12atfcu-3 hixidz"
            id="manager-response-mandatory-for-manager-review:-"
            labelposition="left"
          >
            <label
              class="sc-975cgx-0 aFzmi sc-12atfcu-1 gBASdL"
              for="manager-response-mandatory-for-manager-review:-"
              labelposition="left"
              labelvariant="h4"
              labelweight="Regular"
            >
              Manager Response Mandatory for Manager Review: 
            </label>
            <div
              class="sc-12atfcu-0 sc-1oxfop0-0 hjFLkE hzRUNj"
            >
              <input
                class="sc-1oxfop0-5 kOEioh sc-tkiih6-30"
                data-testid="toggleManagerReview"
                id="manager-response-mandatory-for-manager-review:-"
                type="checkbox"
              />
              <div
                class="sc-1oxfop0-1 kTppyc"
              >
                <div
                  class="sc-1oxfop0-4 fOqcXx"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-tkiih6-29 sc-1porr9n-9 fWtWBE ldQZxM"
        >
          <div
            class="sc-12atfcu-3 hixidz"
            id="employees-can-submit-suggestions-anonymously:-"
            labelposition="left"
          >
            <label
              class="sc-975cgx-0 aFzmi sc-12atfcu-1 gBASdL"
              for="employees-can-submit-suggestions-anonymously:-"
              labelposition="left"
              labelvariant="h4"
              labelweight="Regular"
            >
              Employees can submit suggestions Anonymously: 
            </label>
            <div
              class="sc-12atfcu-0 sc-1oxfop0-0 hjFLkE hzRUNj"
            >
              <input
                class="sc-1oxfop0-5 kOEioh sc-tkiih6-30"
                data-testid="toggleAnonymousSuggestion"
                id="employees-can-submit-suggestions-anonymously:-"
                type="checkbox"
              />
              <div
                class="sc-1oxfop0-1 kTppyc"
              >
                <div
                  class="sc-1oxfop0-4 fOqcXx"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-tkiih6-29 sc-1porr9n-9 fWtWBE ldQZxM"
        >
          <div
            class="sc-12atfcu-3 hixidz"
            id="send-bi-weekly-feedback-reminders-to-slack-channel:-"
            labelposition="left"
          >
            <label
              class="sc-975cgx-0 aFzmi sc-12atfcu-1 gBASdL"
              for="send-bi-weekly-feedback-reminders-to-slack-channel:-"
              labelposition="left"
              labelvariant="h4"
              labelweight="Regular"
            >
              Send Bi-weekly feedback reminders to slack channel: 
            </label>
            <div
              class="sc-12atfcu-0 sc-1oxfop0-0 hjFLkE hzRUNj"
            >
              <input
                checked=""
                class="sc-1oxfop0-5 kOEioh sc-tkiih6-30"
                data-testid="isBiWeeklyFeedbackReminderEnabled"
                id="send-bi-weekly-feedback-reminders-to-slack-channel:-"
                type="checkbox"
              />
              <div
                class="sc-1oxfop0-1 kTppyc"
              >
                <div
                  class="sc-1oxfop0-4 fOqcXx"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-murh6i-6 sc-1porr9n-7 fyeShk ehPJrx"
      >
        <div
          class="sc-1porr9n-2 dsAHql"
        >
          <svg
            class="sc-1pf3x8a-0 kuVwQe"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M18 4h1c1.1 0 2 .9 2 2v14c0 1.1-.9 2-2 2H5a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h1V3c0-.55.45-1 1-1s1 .45 1 1v1h8V3c0-.55.45-1 1-1s1 .45 1 1v1zM8 11h8c.55 0 1 .45 1 1s-.45 1-1 1H8c-.55 0-1-.45-1-1s.45-1 1-1zm10 9H6c-.55 0-1-.45-1-1V9h14v10c0 .55-.45 1-1 1zM8 15h5c.55 0 1 .45 1 1s-.45 1-1 1H8c-.55 0-1-.45-1-1s.45-1 1-1z"
              fill="#000"
              fill-opacity=".54"
              fill-rule="evenodd"
            />
          </svg>
          <h3
            class="sc-1ntgl3l-0 lhJlYH sc-tkiih6-22 isRtxH"
            textalign="initial"
            textcolor="inherit"
            textvariant="h3"
            textweight="Strong"
          >
            Notes
          </h3>
        </div>
        <ul
          class="sc-1xeus4s-1 hIctrP"
        >
          <li
            class="sc-1xeus4s-2 jIFqE"
          >
            <span
              class="sc-1ntgl3l-0 hGagSU"
              style="font-size: 14px;"
              textalign="initial"
              textcolor="inherit"
            >
              <strong>
                Manager Response Mandatory:
              </strong>
               When enabled, managers must provide both a review and a rating during the review process. If disabled, only a rating is required during the review process.
            </span>
          </li>
          <li
            class="sc-1xeus4s-2 jIFqE"
          >
            <span
              class="sc-1ntgl3l-0 hGagSU"
              style="font-size: 14px;"
              textalign="initial"
              textcolor="inherit"
            >
              <strong>
                Anonymous Suggestions:
              </strong>
               When enabled, employees have the option to submit their suggestions/feedback to Suggestion Box anonymously, ensuring privacy and encouraging open feedback.
            </span>
          </li>
          <li
            class="sc-1xeus4s-2 jIFqE"
          >
            <span
              class="sc-1ntgl3l-0 hGagSU"
              style="font-size: 14px;"
              textalign="initial"
              textcolor="inherit"
            >
              <strong>
                Bi-weekly reminders:
              </strong>
               When enabled, reminders will be sent to the integrated slack channel every alternate Monday at 12 pm to share feedback and keep 1:1 conversations active.
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="sc-1porr9n-0 gWazlW"
    >
      <div
        class="sc-1porr9n-1 cnhRJl"
      >
        <h4
          class="sc-1ntgl3l-0 fgmHgf sc-1porr9n-8 iWBSaF"
          textalign="initial"
          textcolor="inherit"
          textvariant="h4"
          textweight="Regular"
        >
          Allowed Domains
        </h4>
        <div
          class="sc-1porr9n-3 dHaVtv"
        >
          <div
            class="sc-17d4s39-0 bCCAZW sc-1porr9n-4 lfbpYT"
            id="medly-textField-input-wrapper"
            minwidth="20rem"
          >
            <div
              class="sc-19px6zp-0 UwvIf"
              disabled=""
              filled="[object Object]"
              fusion="[object Object]"
              minrows="1"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <div
                class="sc-lhr232-0 btNOKw"
                variant="outlined"
              >
                <input
                  aria-describedby="medly-textField-helper-text"
                  class="sc-hb99wz-0 jaEbhp"
                  data-testid="domain"
                  disabled=""
                  errortext=""
                  id="medly-textField-input"
                  inputsize="M"
                  name="name"
                  placeholder=" "
                  type="text"
                  value="@test.com"
                  variant="outlined"
                />
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="medly-textField-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  Domain Name
                </label>
              </div>
            </div>
            
          </div>
        </div>
        <div
          class="sc-1porr9n-3 dHaVtv"
        >
          <div
            class="sc-17d4s39-0 bCCAZW sc-1porr9n-4 lfbpYT"
            id="medly-textField-input-wrapper"
            minwidth="20rem"
          >
            <div
              class="sc-19px6zp-0 UwvIf"
              disabled=""
              filled="[object Object]"
              fusion="[object Object]"
              minrows="1"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <div
                class="sc-lhr232-0 btNOKw"
                variant="outlined"
              >
                <input
                  aria-describedby="medly-textField-helper-text"
                  class="sc-hb99wz-0 jaEbhp"
                  data-testid="domain"
                  disabled=""
                  errortext=""
                  id="medly-textField-input"
                  inputsize="M"
                  name="name"
                  placeholder=" "
                  type="text"
                  value="@gmail.com"
                  variant="outlined"
                />
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="medly-textField-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  Domain Name
                </label>
              </div>
            </div>
            
          </div>
          <button
            class="sc-tz4tqh-13 dgQxHm"
            title=""
          >
            <svg
              class="sc-1pf3x8a-0 cKNfEU"
              data-testid="deleteDomainButton"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.79 3.29l.71.71H18c.55 0 1 .45 1 1s-.45 1-1 1H6c-.55 0-1-.45-1-1s.45-1 1-1h2.5l.71-.71c.18-.18.44-.29.7-.29h4.18c.26 0 .52.11.7.29zM6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v10zm3.17-7.83a.996.996 0 0 1 1.41 0L12 12.59l1.42-1.42a.996.996 0 1 1 1.41 1.41L13.41 14l1.42 1.42a.996.996 0 1 1-1.41 1.41L12 15.41l-1.42 1.42a.996.996 0 1 1-1.41-1.41L10.59 14l-1.42-1.42a.996.996 0 0 1 0-1.41z"
                fill="#000"
                fill-opacity=".54"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="sc-1porr9n-3 dHaVtv"
        >
          <div
            class="sc-17d4s39-0 bCCAZW sc-1porr9n-4 lfbpYT"
            id="medly-textField-input-wrapper"
            minwidth="20rem"
          >
            <div
              class="sc-19px6zp-0 UwvIf"
              disabled=""
              filled="[object Object]"
              fusion="[object Object]"
              minrows="1"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <div
                class="sc-lhr232-0 btNOKw"
                variant="outlined"
              >
                <input
                  aria-describedby="medly-textField-helper-text"
                  class="sc-hb99wz-0 jaEbhp"
                  data-testid="domain"
                  disabled=""
                  errortext=""
                  id="medly-textField-input"
                  inputsize="M"
                  name="name"
                  placeholder=" "
                  type="text"
                  value="@scalereal.com"
                  variant="outlined"
                />
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="medly-textField-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  Domain Name
                </label>
              </div>
            </div>
            
          </div>
          <button
            class="sc-tz4tqh-13 dgQxHm"
            title=""
          >
            <svg
              class="sc-1pf3x8a-0 cKNfEU"
              data-testid="deleteDomainButton"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.79 3.29l.71.71H18c.55 0 1 .45 1 1s-.45 1-1 1H6c-.55 0-1-.45-1-1s.45-1 1-1h2.5l.71-.71c.18-.18.44-.29.7-.29h4.18c.26 0 .52.11.7.29zM6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v10zm3.17-7.83a.996.996 0 0 1 1.41 0L12 12.59l1.42-1.42a.996.996 0 1 1 1.41 1.41L13.41 14l1.42 1.42a.996.996 0 1 1-1.41 1.41L12 15.41l-1.42 1.42a.996.996 0 1 1-1.41-1.41L10.59 14l-1.42-1.42a.996.996 0 0 1 0-1.41z"
                fill="#000"
                fill-opacity=".54"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="sc-1nys386-13"
        >
          <h4
            class="sc-1ntgl3l-0 igfIhV sc-1porr9n-5 gquEcb"
            textalign="center"
            textcolor="inherit"
            textvariant="h4"
          >
            + Add Domain
          </h4>
        </div>
        <div
          class="sc-1porr9n-6 hyKqtN"
        >
          <button
            class="sc-1k2druj-0 eLgfbc"
            disabled=""
            edges="square"
            type="button"
            variant="solid"
          >
            <span
              class="sc-1ntgl3l-0 eOsHHD"
              textalign="initial"
              textcolor="inherit"
              textvariant="button1"
            >
              Save
            </span>
          </button>
        </div>
      </div>
      <div
        class="sc-murh6i-6 sc-1porr9n-7 fyeShk ehPJrx"
      >
        <div
          class="sc-1porr9n-2 dsAHql"
        >
          <svg
            class="sc-1pf3x8a-0 kuVwQe"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M18 4h1c1.1 0 2 .9 2 2v14c0 1.1-.9 2-2 2H5a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h1V3c0-.55.45-1 1-1s1 .45 1 1v1h8V3c0-.55.45-1 1-1s1 .45 1 1v1zM8 11h8c.55 0 1 .45 1 1s-.45 1-1 1H8c-.55 0-1-.45-1-1s.45-1 1-1zm10 9H6c-.55 0-1-.45-1-1V9h14v10c0 .55-.45 1-1 1zM8 15h5c.55 0 1 .45 1 1s-.45 1-1 1H8c-.55 0-1-.45-1-1s.45-1 1-1z"
              fill="#000"
              fill-opacity=".54"
              fill-rule="evenodd"
            />
          </svg>
          <h3
            class="sc-1ntgl3l-0 lhJlYH sc-tkiih6-22 isRtxH"
            textalign="initial"
            textcolor="inherit"
            textvariant="h3"
            textweight="Strong"
          >
            Notes
          </h3>
        </div>
        <span
          class="sc-1ntgl3l-0 hGagSU"
          style="padding-top: 10px; font-size: 14px;"
          textalign="initial"
          textcolor="inherit"
        >
          <strong>
            Allowed Domain: 
          </strong>
          You can add domains if you wish to allow consultants or contractors from other organisation domains to be a part of your 360-degree feedback and performance review processes
        </span>
      </div>
    </div>
  </main>
</div>
`;
